server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTable.java

```diff
@@ -115,15 +113,15 @@ public TooManyMappingException(String string) {
      */
     Map<MappingSource, Mappings> getAllMappings() throws RecipientRewriteTableException;
 
-    default List<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
+    default Stream<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
         Preconditions.checkArgument(listSourcesSupportedType.contains(mapping.getType()),
             String.format("Not supported mapping of type %s", mapping.getType()));
 
         return getAllMappings()
             .entrySet().stream()
             .filter(entry -> entry.getValue().contains(mapping))
             .map(Map.Entry::getKey)
-            .collect(Guavate.toImmutableList());
+            .sorted(Comparator.comparing(MappingSource::asMailAddressString));
```

Comment:
If we sort it, then we have to process all sources, which defeats the performance optimisation enabled by the stream optimisation.

Maybe we need to sort 'only before collecting' and delegate that responsibility to the caller?